hail/python/hail/vds/variant_dataset.py

```diff
@@ -326,6 +356,38 @@ def error(msg):
     def _same(self, other: 'VariantDataset'):
         return self.reference_data._same(other.reference_data) and self.variant_data._same(other.variant_data)
 
+    @staticmethod
+    def _add_len(rd):
+        if 'LEN' in rd.entry:
+            return rd
+        if 'END' not in rd.entry:
+            raise ValueError('Need `END` to compute `LEN` in reference data')
+        return rd.annotate_entries(LEN=rd.END - rd.locus.position + 1)
```

Comment:
Optional - it might just be me but I took a few seconds to parse this... I think maybe because the two `if`s were mixing positive and negative assertions and the default `return` is actually the positive case of the second `if`. Does that make any sense?

In other words, I think this (and the ones below) might be a little easier to follow if the second `if` were also a "if (positive)" case, eg:
```py
        if 'LEN' in rd.entry:
            return rd
        elif 'END' in rd.entry:
            return rd.annotate_entries(LEN=rd.END - rd.locus.position + 1)
        else:
            raise ValueError('Need `END` to compute `LEN` in reference data')
```

Reply:
This is clearer. I like this better.